app/src/main/java/me/rocka/fcitx5test/ui/common/DialogSeekBarPreference.kt

```diff
@@ -0,0 +1,112 @@
+ * @property min The minimum value of the seek bar. Must not be greater or equal than [max].
+ * @property max The maximum value of the seek bar. Must not be lesser or equal than [min].
+ * @property step The step in which the seek bar increases per move. If the provided value is less
+ *  than 1, 1 will be used as step.
+ * @property unit The unit to show after the value. Set to an empty string to disable this feature.
+ */
+class DialogSeekBarPreference : Preference {
+    var defaultValue: Int = 0
+    var min: Int = 0
+    var max: Int = 100
+    var step: Int = 1
+    var unit: String = ""
+
+    private val currentValue: Int
+        get() = preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
+
+    @Suppress("unused")
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) :
+            this(context, attrs, R.attr.preferenceStyle)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
+            super(context, attrs, defStyleAttr) {
+        onPreferenceClickListener = OnPreferenceClickListener {
+            showSeekBarDialog()
+            true
+        }
+    }
+
+    override fun onAttachedToHierarchy(preferenceManager: PreferenceManager?) {
+        super.onAttachedToHierarchy(preferenceManager)
+        summary = getTextForValue(currentValue)
+    }
+
+    /**
+     * Generates the text for the given [value] and adds the defined [unit] at the end.
+     */
+    private fun getTextForValue(value: Any): String {
+        return if (value !is Int) {
+            "??$unit"
+        } else {
+            value.toString() + unit
+        }
+    }
+
+    /**
+     * Shows the seek bar dialog.
+     */
+    private fun showSeekBarDialog() {
+        val dialogView = DialogSeekBarBinding.inflate(LayoutInflater.from(context))
+        val initValue = currentValue
+        dialogView.seekBar.max = actualValueToSeekBarProgress(max)
+        dialogView.seekBar.progress = actualValueToSeekBarProgress(initValue)
+        dialogView.seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+                dialogView.seekBarValue.text = getTextForValue(seekBarProgressToActualValue(progress))
+            }
+            override fun onStartTrackingTouch(seekBar: SeekBar?) {}
+            override fun onStopTrackingTouch(seekBar: SeekBar?) {}
+        })
+        dialogView.seekBarValue.text = getTextForValue(initValue)
+        AlertDialog.Builder(context).apply {
+            setTitle(this@DialogSeekBarPreference.title)
+            setView(dialogView.root)
+            setPositiveButton(android.R.string.ok) { _, _ ->
+                val actualValue = seekBarProgressToActualValue(dialogView.seekBar.progress)
+                preferenceDataStore?.putInt(key, actualValue)
+                summary = getTextForValue(currentValue)
```

Comment:
```suggestion
                if (callChangeListener(actualValue)) {
                    preferenceDataStore?.putInt(key, actualValue)
                    summary = getTextForValue(actualValue)
                }
```

According to the documentation, onPreferenceChangedListener should be called before updating internal state:

https://developer.android.com/reference/androidx/preference/Preference#setOnPreferenceChangeListener(androidx.preference.Preference.OnPreferenceChangeListener)

... and `callChangeListener` mentioned here:

https://developer.android.com/reference/androidx/preference/Preference#onClick()